src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -14,6 +14,13 @@ internal sealed class ConnectionPool
 		{
 			cancellationToken.ThrowIfCancellationRequested();
 
+			// if all sessions are used, see if any have been leaked and can be recovered
+			if (m_sessionSemaphore.CurrentCount == 0)
+			{
+				GC.Collect();
+				RecoverLeakedSessions();
```

Comment:
The other option would be just to delete this whole `if` statement. and hope that connections are being leaked slowly enough that they've been GCed and cleaned up by the background task before the pool is emptied. 

This would stop "punishing" clients who are correctly disposing their connections but are legitimately hitting an empty pool.